test/tools/jsontests/StateTests.cpp

```diff
@@ -142,15 +142,28 @@ class GeneralTestFixture
 public:
 	GeneralTestFixture()
 	{
-		test::StateTestSuite suite;
-		string casename = boost::unit_test::framework::current_test_case().p_name;
-		if (casename == "stQuadraticComplexityTest" && !test::Options::get().all)
-		{
-			std::cout << "Skipping " << casename << " because --all option is not specified.\n";
-			return;
-		}
-		suite.runAllTestsInFolder(casename);
-	}
+        test::StateTestSuite suite;
+        string casename = boost::unit_test::framework::current_test_case().p_name;
+
+        // Mark all test folders to run
+        static bool registerTestFolders = false;
```

Comment:
Why is this needed? Can you instead call `registerTestFolders` for a parent path once from the level above, like from the main function?

If not, better to make this check inside `registerTestFolders`